src/app-layer-ssl.c

```diff
@@ -1431,6 +1437,10 @@ static int EnsureRecordSpace(SSLStateConnp *curr_connp, const uint8_t * const in
         SCLogDebug("cert_len unknown still, create small buffer to start");
         certs_len = 256;
     }
+    // Limit in a first time allocation for very large certificates
```

Comment:
I'm struggling a bit with understanding how this will work for actual large records. Are we getting called again in the future? A casual read of the code suggests it actually expects to be called with the full record, incl the starting bytes. If we'd have a record size that is larger than the initial input size, which will almost always be true if it is more than say 4k, do we still handle things correctly? I guess I'd love to see some test cases for this condition.

Reply:
btw my lack of understanding is more about the original code, but its relevant to judging this PR

Reply:
> btw my lack of understanding is more about the original code

Oh, I was indeed assuming you were familiar with it...
One of my goals for this PR is to have minimal changes (until we get a rust tls parser)

> Are we getting called again in the future?

Yes ! this function `EnsureRecordSpace` gets called for every packet  which contains bytes of the certificate even if only the 3 first bytes are useful... 🙄

> A casual read of the code suggests it actually expects to be called with the full record

I do not see this... Where do you see this ?

Reply:
I think I see my error. On consecutive calls it does use the buffered data to get the length.

Reply:
> On consecutive calls it does use the buffered data to get the length.

Indeed...
So, do you expect something from me ?